pkg/cli/cli.go

```diff
@@ -290,7 +320,18 @@ func NewCmd[T any, PtrT *T](
 	cmd.PersistentFlags().String("grant-principal-type", "", "The resource type of the principal to grant the entitlement to ($BATON_GRANT_PRINCIPAL_TYPE)")
 	cmd.MarkFlagsRequiredTogether("grant-entitlement", "grant-principal", "grant-principal-type")
 	cmd.PersistentFlags().String("revoke-grant", "", "The grant to revoke ($BATON_REVOKE_GRANT)")
-	cmd.MarkFlagsMutuallyExclusive("grant-entitlement", "revoke-grant")
+
```

Comment:
I wonder if we should consider moving these provisioning things to a sub command

Reply:
I agree, but that would break backwards compatibility and I don't want to do that in this pull request. Ideally I think the command line params should look something like this:

```
baton-connector --provision-action=grant --entitlement=entitlement_id --resource=resource_id --resource-type=resource_type
baton-connector --provision-action=revoke --grant=grant_id
baton-connector --provision-action=delete --resource-id=resource_id --resource-type=resource_type
baton-connector --provision-action=create-account --login=login --email=email
baton-connector --provision-action=rotate-credentials --resource-id=resource_id --resource-type=resource_type
```